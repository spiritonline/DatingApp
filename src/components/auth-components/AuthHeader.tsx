import { useColorScheme } from 'react-native';
import styled from 'styled-components/native';
import { ThemeProps } from '../../utils/styled-components';

interface AuthHeaderProps {
  title: string;
  onBack: () => void;
}



export default function AuthHeader({ title, onBack }: AuthHeaderProps) {
  const colorScheme = useColorScheme();
  const isDark = colorScheme === 'dark';

  return (
    <Container>
      <BackButton 
        onPress={onBack}
        accessibilityRole="button"
        accessibilityLabel="Back"
        accessibilityHint="Go back to previous screen"
        testID="back-button"
      >
        <BackArrow isDark={isDark}>‚Üê</BackArrow>
      </BackButton>
      <Title isDark={isDark}>{title}</Title>
      <Spacer />
    </Container>
  );
}

const Container = styled.View`
  width: 100%;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 16px 20px;
`;

const BackButton = styled.TouchableOpacity`
  padding: 8px;
`;

const BackArrow = styled.Text<ThemeProps>`
  font-size: 24px;
  color: ${(props: ThemeProps) => (props.isDark ? '#e0e0e0' : '#333')};
`;

const Title = styled.Text<ThemeProps>`
  font-size: 22px;
  font-weight: bold;
  color: ${(props: ThemeProps) => (props.isDark ? '#e0e0e0' : '#333')};
`;

const Spacer = styled.View`
  width: 24px;
`;
