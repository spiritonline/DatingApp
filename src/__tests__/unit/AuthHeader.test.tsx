import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import AuthHeader from '../../components/auth-components/AuthHeader';

describe('AuthHeader', () => {
  const mockOnBack = jest.fn();
  const title = 'Test Header';

  beforeEach(() => {
    mockOnBack.mockClear();
  });

  it('renders correctly with the provided title', () => {
    const { getByText } = render(<AuthHeader title={title} onBack={mockOnBack} />);
    expect(getByText(title)).toBeTruthy();
  });

  it('calls onBack when back button is pressed', () => {
    const { getByTestId } = render(<AuthHeader title={title} onBack={mockOnBack} />);
    const backButton = getByTestId('back-button');
    
    fireEvent.press(backButton);
    
    expect(mockOnBack).toHaveBeenCalledTimes(1);
  });
  
  it('matches snapshot', () => {
    const { toJSON } = render(<AuthHeader title={title} onBack={mockOnBack} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
