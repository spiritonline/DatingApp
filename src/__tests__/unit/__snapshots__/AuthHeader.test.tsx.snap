// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHeader matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 16,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 16,
      "width": "100%",
    }
  }
>
  <View
    accessibilityHint="Go back to previous screen"
    accessibilityLabel="Back"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
    testID="back-button"
  >
    <Text
      isDark={false}
      style={
        {
          "color": "#333",
          "fontSize": 24,
        }
      }
    >
      ‚Üê
    </Text>
  </View>
  <Text
    isDark={false}
    style={
      {
        "color": "#333",
        "fontSize": 22,
        "fontWeight": "bold",
      }
    }
  >
    Test Header
  </Text>
  <View
    style={
      {
        "width": 24,
      }
    }
  />
</View>
`;
